### ISSUE TITLE
Update agency matching form

### ISSUE THREAD

**Issue by fditrapani** [Automattic/wp-calypso#102060]
## Problem

The agency matching multi-step form looks really rough:

<img width="1840" alt="Image" src="https://github.com/user-attachments/assets/784e6087-ac77-42e1-a3f8-5a91d8c2250e" />

<img width="857" alt="Image" src="https://github.com/user-attachments/assets/fa332415-bd52-476f-850e-26e3247ece31" />

<img width="1840" alt="Image" src="https://github.com/user-attachments/assets/3911a2b8-dd83-4fe1-b29f-90d2801eaa9a" />

![Image](https://github.com/user-attachments/assets/9273b462-b5e4-4e7d-88f3-28a58f7ccd2a)

- Missing branding + navigation
- Lots of boxes
- Some weird indents and misaligned fields
- The footer has a border around it
- Broken-looking back buttons
- Lots of fields to complete with lots of text to parse
- A generic broken-looking success page

This could use a redesign

## Where
https://wordpress.com/website-design-service/get-started

Accessed from: 
- Visit https://wordpress.com/website-design-service
- Go to the plans and select match me with an agency

**Comment by fditrapani** [comment]
We unfortunately don't have any tracks events to show how often this is getting used.

**Comment by fditrapani** [comment]
This is being replaced with another form: https://a4aproduct.wordpress.com/2025/03/19/call-for-testing-hire-an-expert-landing-pages/
